anomaly-benchmarking/generate-datasets/script.whizzml

```diff
@@ -131,16 +153,16 @@
 ;; possible. Generates the max possible datasets (limit 40) at those
 ;; frequencies. Returns a list of datasets.
 
-(define (relative-frequency ds-id normal-rows)
+(define (relative-frequency ds-id normal-rows replicates freq-list)
   (let (ds (fetch ds-id)
         rows (ds "rows")
         ds-name (ds "name")
         my-tags (ds "tags")
         max-freq (/ rows (+ rows normal-rows)))
-    (for (freq [0.001 0.005 0.01 0.05 0.1])
+    (for (freq freq-list)
       (let (k (/ (* freq normal-rows) (- 1 freq)))
         (when (> max-freq freq)
-          (repeatedly (min 10 (floor (/ rows k))) 
+          (repeatedly (min replicates (floor (/ rows k))) 
```

Comment:
`k` is the number of anomalous points, so it would be strange to test an anomaly detector by purposely giving it a dataset without anomalous points. It's not completely ridiculous, however.  I'll put in a catch for the divide by zero error.